src/DataStructures/DataVector.cpp

```diff
@@ -12,8 +12,11 @@ DataVector::DataVector(const size_t size, const double value)
       owned_data_(size_, value),
       data_(owned_data_.data(), size_) {}
 
-DataVector::DataVector(std::initializer_list<double> list)
-    : size_(list.size()), owned_data_(list), data_(owned_data_.data(), size_) {}
+template <class T, Requires<std::is_same<std::decay_t<T>, double>::value>>
```

Comment:
This Requires `decay_t<T>` to be double, but it is only usable for T=double because of the explicit instantiation.  Can T ever be something else that decays to double?

Reply:
If we want to be safe against potentially weird things with function pointers we can change to `std::remove_cv_t<std::remove_reference_t<T>>`, that would be fine. The other possible case is `float` -> `double`, which we may want allow. Thoughts?